packet/conn.go

```diff
@@ -80,7 +80,7 @@ func (c *Conn) ReadPacketTo(w io.Writer) error {
 	header := []byte{0, 0, 0, 0}
 
 	if _, err := io.ReadFull(c.Conn, header); err != nil {
-		return ErrBadConn
+		return errors.Wrapf(ErrBadConn, "io.ReadFull(header) failed. err %v", err)
```

Comment:
preserving dummy `ErrBadConn` here, because code in canal/canal.go, failover/server.go and driver/driver.go is checking for it by applying errors.Cause